From 3b1adb0d79a571063f4fb24c5cbd0553584ab34d Mon Sep 17 00:00:00 2001
From: Jianqun Xu <jay.xu@rock-chips.com>
Date: Tue, 5 Jan 2021 10:09:48 +0800
Subject: [PATCH] ARM: mm: fix virtual kernel memory layout display format

Before this patch:
[    0.000000] Virtual kernel memory layout:
[    0.000000]     vector  : 0xffff0000 - 0xffff1000   (   4 kB)
[    0.000000]     fixmap  : 0xffc00000 - 0xfff00000   (3072 kB)
[    0.000000]     vmalloc : 0xf0800000 - 0xff800000   ( 240 MB)
[    0.000000]     lowmem  : 0xb0000000 - 0xf0000000   (1024 MB)
[    0.000000]     pkmap   : 0xafe00000 - 0xb0000000   (   2 MB)
[    0.000000]     modules : 0xaf000000 - 0xafe00000   (  14 MB)
[    0.000000]       .text : 0x(ptrval) - 0x(ptrval)   (9184 kB)
[    0.000000]       .init : 0x(ptrval) - 0x(ptrval)   (1024 kB)
[    0.000000]       .data : 0x(ptrval) - 0x(ptrval)   ( 331 kB)
[    0.000000]        .bss : 0x(ptrval) - 0x(ptrval)   ( 896 kB)

With this patch:
[    0.000000] Virtual kernel memory layout:
[    0.000000]     vector  : 0xffff0000 - 0xffff1000   (   4 kB)
[    0.000000]     fixmap  : 0xffc00000 - 0xfff00000   (3072 kB)
[    0.000000]     vmalloc : 0xf0800000 - 0xff800000   ( 240 MB)
[    0.000000]     lowmem  : 0xb0000000 - 0xf0000000   (1024 MB)
[    0.000000]     pkmap   : 0xafe00000 - 0xb0000000   (   2 MB)
[    0.000000]     modules : 0xaf000000 - 0xafe00000   (  14 MB)
[    0.000000]       .text : 0xb0008000 - 0xb0900000   (9184 kB)
[    0.000000]       .init : 0xb0c00000 - 0xb0d00000   (1024 kB)
[    0.000000]       .data : 0xb0d00000 - 0xb0d52d4c   ( 332 kB)
[    0.000000]        .bss : 0xb0d52d4c - 0xb0e329b0   ( 896 kB)

Change-Id: I3c77ff89dc5320213f4ff1e0a89b3c688f1d0638
Signed-off-by: Jianqun Xu <jay.xu@rock-chips.com>
---
 arch/arm/mm/init.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/arch/arm/mm/init.c b/arch/arm/mm/init.c
index ade055556a7b..9720794fe289 100644
--- a/arch/arm/mm/init.c
+++ b/arch/arm/mm/init.c
@@ -543,10 +543,10 @@ void __init mem_init(void)
 #ifdef CONFIG_MODULES
 			"    modules : 0x%08lx - 0x%08lx   (%4ld MB)\n"
 #endif
-			"      .text : 0x%p" " - 0x%p" "   (%4td kB)\n"
-			"      .init : 0x%p" " - 0x%p" "   (%4td kB)\n"
-			"      .data : 0x%p" " - 0x%p" "   (%4td kB)\n"
-			"       .bss : 0x%p" " - 0x%p" "   (%4td kB)\n",
+			"      .text : 0x%px" " - 0x%px" "   (%4td kB)\n"
+			"      .init : 0x%px" " - 0x%px" "   (%4td kB)\n"
+			"      .data : 0x%px" " - 0x%px" "   (%4td kB)\n"
+			"       .bss : 0x%px" " - 0x%px" "   (%4td kB)\n",
 
 			MLK(VECTORS_BASE, VECTORS_BASE + PAGE_SIZE),
 #ifdef CONFIG_HAVE_TCM
-- 
2.35.3

